esmvaltool/diag_scripts/kcs/resampling.py

```diff
@@ -127,50 +132,74 @@ def select_final_subset(combinations, n=8):
 
 
 def main(cfg):
+    """Resample the model of interest.
+
+    Step 0: Read the data, extract segmented subsets for both the control
+    and future periods, and precompute seasonal means for each segment.
+
+    Step 1a: Get 1000 combinations for the control period.
+    These samples should have the same mean winter precipitation as the
+    overall mean of the x ensemble members.
+
+    Step 1b: Get 1000 combinations for the future period.
+    The target value is a relative change wrt,
+    the overall mean of the control period.
+
+    Step 2a
+    Step 2b
+    Step 3
+    """
 
-    # Step 0: Read the data, extract segmented subsets for both the control
-    # and future periods, and precompute seasonal means for each segment.
+    # Start step 0:
+    # Read the data
+    # TODO define a function to get input data
     dataset_dicts = cfg['input_data'].values()
     target_model_metadata = select_metadata(
-        dataset_dicts, dataset=cfg['target_model'])
+        dataset_dicts, dataset=cfg['target_model']
+    )
     files = [metadata['filename'] for metadata in target_model_metadata]
-    dataset = xr.open_mfdataset(
-        files, concat_dim='ensemble_member', combine='nested')[['pr', 'tas']]
-    segments_season_means = {}
+
+    # TODO nor working, it needs another dimension for variables!
```

Comment:
Xarray, differently then Iris, doesn't need dimensions for variables. An `xarray.DataArray` object can represent only one variables, but multiple variables can be combined into an `xarray.Dataset` object.